agent/api/task.go

```diff
@@ -423,9 +481,56 @@ func (task *Task) dockerHostConfig(container *Container, dockerContainerMap map[
 		}
 	}
 
+	// Determine if network mode should be overridden and override it if needed
+	ok, networkMode := task.shouldOverrideNetworkMode(container, dockerContainerMap)
+	if !ok {
+		return hostConfig, nil
+	}
+	hostConfig.NetworkMode = networkMode
 	return hostConfig, nil
 }
 
+// shouldOverrideNetworkMode returns true if the network mode of the container needs
+// to be overridden. It also returns the override string in this case. It returns
+// false otherwise
+func (task *Task) shouldOverrideNetworkMode(container *Container, dockerContainerMap map[string]*DockerContainer) (bool, string) {
```

Comment:
I think I would prefer just a `getNetworkMode` that always returns a network mode which is either the overridden one or the originally specified one.  I don't think we need the `ok` return if this function can do everything.